quartz_solar_forecast/forecast.py

```diff
@@ -30,6 +26,110 @@ def run_forecast(site: PVSite, ts: datetime | str = None, nwp_source: str = "ico
     pv_xr = make_pv_data(site=site, ts=ts)
 
     # load and run models
-    pred_df = forecast_v1(nwp_source, nwp_xr, pv_xr, ts)
+    pred_df = forecast_v1(nwp_source, nwp_xr, pv_xr, ts, model=model)
 
     return pred_df
+
+
+def predict_tryolabs(
+    site: PVSite, model=None, ts: datetime | str = None):
+    """Run the forecast with the tryolabs model"""
+    solar_power_predictor = TryolabsSolarPowerPredictor(model=model)
+
+    if ts is None:
+        start_date = pd.Timestamp.now().strftime("%Y-%m-%d")
+        start_time = pd.Timestamp.now().floor("15min")
+    else:
+        start_date = pd.Timestamp(ts).strftime("%Y-%m-%d")
+        start_time = pd.Timestamp(ts).floor("15min")
+  
+    end_time = start_time + pd.Timedelta(hours=48)
+
+    predictions = solar_power_predictor.predict_power_output(
+        latitude=site.latitude,
+        longitude=site.longitude,
+        start_date=start_date,
+        kwp=site.capacity_kwp,
+        orientation=site.orientation,
+        tilt=site.tilt,
+    )
+
+    if predictions is not None:
+        predictions = predictions[
+            (predictions["date"] >= start_time) & (predictions["date"] < end_time)
+        ]
+        predictions = predictions.reset_index(drop=True)
+        predictions.set_index("date", inplace=True)
+        print("Predictions finished.")
+        return predictions
+
+
+def download_model(filename: str, file_id: str):
```

Comment:
I would push all the tryolabs specific stuff up to forecasts/v2.py and then we can keep this file very simple